docs/aws/audit/redshiftmonitoring/rules/default_master_username_remediation.mdx

```diff
@@ -1,6 +1,248 @@
+
+1. **Navigate to Redshift Dashboard:**
+   - Open the AWS Management Console.
+   - In the search bar, type "Redshift" and select "Amazon Redshift" from the dropdown.
+
+2. **Create a New Cluster:**
+   - Click on the "Create cluster" button.
+   - Fill in the necessary details for your new cluster.
+
+3. **Specify Master Username:**
+   - In the "Database configurations" section, locate the "Master user settings."
+   - Enter a custom master username in the "Master user name" field. Avoid using the default username "awsuser."
+
+4. **Review and Launch:**
+   - Continue with the remaining configuration steps.
+   - Review your settings and click "Create cluster" to launch the cluster with the custom master username.
+
+By following these steps, you ensure that the Redshift cluster does not use the default master username, enhancing the security of your database.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the use of the default master username in Amazon Redshift using AWS CLI, follow these steps:
+
+1. **Create a New Redshift Cluster with a Custom Master Username:**
+   When creating a new Redshift cluster, specify a custom master username using the `--master-username` parameter.
+
+   ```sh
+   aws redshift create-cluster \
+       --cluster-identifier my-cluster \
+       --node-type dc2.large \
+       --master-username customadmin \
+       --master-user-password MySecurePassword123 \
+       --cluster-type single-node
+   ```
+
+2. **Modify an Existing Redshift Cluster to Change the Master Username:**
+   Note that AWS does not allow changing the master username of an existing cluster directly. Instead, you need to create a new cluster with the desired username and migrate your data.
+
+3. **Use AWS CloudFormation to Enforce Custom Master Username:**
+   Define a CloudFormation template that specifies a custom master username. This ensures that any Redshift cluster created using this template will not use the default master username.
+
+   ```yaml
+   Resources:
+     MyRedshiftCluster:
+       Type: "AWS::Redshift::Cluster"
+       Properties:
+         ClusterIdentifier: "my-cluster"
+         NodeType: "dc2.large"
+         MasterUsername: "customadmin"
+         MasterUserPassword: "MySecurePassword123"
+         ClusterType: "single-node"
+   ```
+
+   Deploy the CloudFormation stack using AWS CLI:
+
+   ```sh
+   aws cloudformation create-stack \
+       --stack-name my-redshift-stack \
+       --template-body file://my-redshift-template.yaml
+   ```
+
+4. **Automate Cluster Creation with a Custom Username Using a Python Script:**
+   Use the AWS SDK for Python (Boto3) to automate the creation of a Redshift cluster with a custom master username.
+
+   ```python
+   import boto3
+
+   redshift = boto3.client('redshift')
+
+   response = redshift.create_cluster(
+       ClusterIdentifier='my-cluster',
+       NodeType='dc2.large',
+       MasterUsername='customadmin',
+       MasterUserPassword='MySecurePassword123',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>